src/main/java/org/jahia/modules/jahiacsrfguard/filters/CsrfGuardJavascriptFilter.java

```diff
@@ -72,6 +76,22 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         ResponseWrapper responseWrapper = new ResponseWrapper(httpResponse);
 
         chain.doFilter(request, responseWrapper);
+        String originalContent = responseWrapper.toString();
+        int length = httpRequest.getContextPath().length();
+        String requestPath = length > 0 ? httpRequest.getRequestURI().substring(length) : httpRequest.getRequestURI();
+
+        // skip filter if not html content type of if path from the request or url resolver not match one of the provided patterns.
+        if (!matchHtmlContentType(responseWrapper) || !(matchPattern(requestPath) || matchUrlResolverPattern(httpRequest))) {
+            logger.debug("Not adding CSRFGuard JS to '{}'", httpRequest.getRequestURI());
+            // In case of files, the response is already committed and cannot be overwritten.
+            if (!response.isCommitted()) {
+                response.getWriter().write(originalContent);
+            } else if (originalContent.length() > 0) {
+                logger.warn("Response from {} has content that could not be written because the response has been already committed", httpRequest.getRequestURI());
```

Comment:
When does this case happen ?

Reply:
This happens with resources files from the modules. I tried to find a way to detect them before processing the chain but never succeed